llama_stack/providers/tests/datasetio/test_datasetio.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+import os
+
+import pytest
+import pytest_asyncio
+
+from llama_stack.apis.datasetio import *  # noqa: F403
+from llama_stack.distribution.datatypes import *  # noqa: F403
+from llama_stack.providers.tests.resolver import resolve_impls_for_test
+
+# How to run this test:
+#
+# 1. Ensure you have a conda with the right dependencies installed. This is a bit tricky
+#    since it depends on the provider you are testing. On top of that you need
+#    `pytest` and `pytest-asyncio` installed.
+#
+# 2. Copy and modify the provider_config_example.yaml depending on the provider you are testing.
+#
+# 3. Run:
+#
+# ```bash
+# PROVIDER_ID=<your_provider> \
+#   PROVIDER_CONFIG=provider_config.yaml \
+#   pytest -s llama_stack/providers/tests/datasetio/test_datasetio.py \
+#   --tb=short --disable-warnings
+# ```
+
+
+@pytest_asyncio.fixture(scope="session")
+async def datasetio_settings():
+    impls = await resolve_impls_for_test(
+        Api.datasetio,
+    )
+    return {
+        "datasetio_impl": impls[Api.datasetio],
+        "datasets_impl": impls[Api.datasets],
+    }
+
+
+async def register_dataset(datasets_impl: Datasets):
+    dataset = DatasetDefWithProvider(
+        identifier="test_dataset",
+        provider_id=os.environ["PROVIDER_ID"],
+        url=URL(
+            uri="https://openaipublic.blob.core.windows.net/simple-evals/mmlu.csv",
```

Comment:
Would I be expected as a client register a dataset that's a CSV? is that the format for all datasets? My understanding was that we are going to normalize all datasets to the same jsonl schema. Are we moving to it?

Reply:
@raghotham I think we should decouple the storage format of the file from the schema the dataset reports. The schema is the sacrosanct part, no?

Reply:
We will normalize datasets once it is loaded. E.g. https://github.com/meta-llama/llama-stack/blob/fa68809a2e6df3b9c57f6f0a62c9f3737cd06063/llama_stack/apis/datasets/datasets.py#L91-L103